pkg/istio-agent/xds-proxy.go

```diff
@@ -384,3 +394,21 @@ func getTLSDialOption(agent *Agent) (grpc.DialOption, error) {
 	transportCreds := credentials.NewTLS(&config)
 	return grpc.WithTransportCredentials(transportCreds), nil
 }
+
+// sendUpstreamWithTimeout sends discovery request with default send timeout.
+func sendUpstreamWithTimeout(ctx context.Context, upstream discovery.AggregatedDiscoveryService_StreamAggregatedResourcesClient,
```

Comment:
woa.. wont this be one too many go routines or timers? esp for EDS ?

Reply:
No. We already do that in ads code https://github.com/istio/istio/blob/b7cff9a0fa8c449917b7903b97d9f503f1384429/pilot/pkg/xds/ads.go#L733 for many connections.
Without this it might block some times https://github.com/istio/istio/blob/b7cff9a0fa8c449917b7903b97d9f503f1384429/pilot/pkg/xds/ads.go#L733